### PR TITLE
*: run clippy for integration tests

### PR DESCRIPTION
**Pull request by tabokie**
### What is changed and how it works?

What's Changed:

Run clippy for integration tests and change script to also run it in CI

### Check List 

Tests
- No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/tikv_kv/src/lib.rs [hunk 1]
```diff
@@ -234,7 +234,9 @@ pub trait Engine: Send + Clone + 'static {
 /// at a specific timestamp. This snapshot is lower-level, a view of the underlying storage.
 pub trait Snapshot: Sync + Send + Clone {
     type Iter: Iterator;
-    type Ext<'a>: SnapshotExt;
+    type Ext<'a>: SnapshotExt
+    where
+        Self: 'a;
```
Comment (sticnarf) [inline components/tikv_kv/src/lib.rs:239]: Is it from a clippy warning? I don't see clippy complaining without the where clause.

Are you running clippy with a more recent toolchain?

Comment (tabokie) [reply components/tikv_kv/src/lib.rs:239]: Yes, I ran cargo fix with nightly-2021-12-17, it seems more robust than our toolchain.

Comment (sticnarf) [reply components/tikv_kv/src/lib.rs:239]: OK, I see this is due to this GAT change https://github.com/rust-lang/rust/issues/87479

**Review by sticnarf — APPROVED** [review]

File: tests/integrations/raftstore/test_bootstrap.rs [hunk 2]
```diff
@@ -184,13 +184,12 @@ fn test_node_switch_api_version() {
...
+            cluster.cfg.storage.set_api_version(to_api);
             if from_api == to_api {
-                // Should start with if there is no api version change
-                cluster.cfg.storage.set_api_version(to_api);
                 cluster.start().unwrap();
                 cluster.shutdown();
             } else {
-                // Should not be able to switch to `to_api`
+                // Should not be able to switch to `to_api`.
                 cluster.cfg.storage.set_api_version(to_api);
```
Comment (youjiali1995) [inline tests/integrations/raftstore/test_bootstrap.rs:193]: Remove this line?
```suggestion
```

**Review by youjiali1995 — APPROVED** [review]

**Pull request merged**